primitives/src/identity_id.rs

```diff
@@ -185,7 +185,21 @@ impl Printable for IdentityId {
 pub struct PortfolioName(pub Vec<u8>);
 
 /// The unique ID of a non-default portfolio.
-pub type PortfolioNumber = u64;
+#[derive(Decode, Encode, Clone, Copy, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct PortfolioNumber(pub u64);
+
+impl Default for PortfolioNumber {
+    fn default() -> Self {
+        Self(1)
```

Comment:
Can you leave the derived `Default` instead of making portfolio 1 the default of `PortfolioNumber`? If we leave `PortfolioNumber(0)` to be the default portfolio number then it would mean undefined portfolio number in storage. Otherwise getting a non-existent portfolio number in storage would return `PortfolioNumber(1)`.

Reply:
> getting a non-existent portfolio number in storage would return PortfolioNumber(1).

True but we do not store `PortfolioNumber` in storage except for `NextPortfolioNumber`. For `NextPortfolioNumber`, `PortfolioNumber(1)` should be returned in case no portfolio exists.

Reply:
FWIW, my last approach (now force pushed) kept the default value as 0 and instead changed `NextPortfolioNumber` to `NumberOfPortfoliosCreated`. However, this needed a storage migration and an API change. Comparing the pros and cons, I think the current approach is better.